processor/src/test/java/org/mapstruct/ap/test/injectionstrategy/spring/annotateWith/decorator/SpringAnnotateWithDecoratorMapperTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.injectionstrategy.spring.annotateWith.decorator;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerDto;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerEntity;
+import org.mapstruct.ap.test.injectionstrategy.shared.Gender;
+import org.mapstruct.ap.test.injectionstrategy.shared.GenderDto;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.WithSpring;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test the behavior of decorators when using AnnotateWith and Spring component model.
+ *
+ * @author Jose Carlos Campanero Ortiz
+ */
+@WithClasses({
+        CustomerDto.class,
+        CustomerEntity.class,
+        Gender.class,
+        GenderDto.class,
+        CustomerSpringComponentQualifiedMapper.class,
+        UpperCaseCustomerNameSpringDefaultMapperDecorator.class,
+        CustomerSpringDefaultMapper.class,
+        GenderSpringDefaultMapper.class
+})
+@IssueKey( "1427" )
+@ComponentScan(basePackageClasses = CustomerSpringDefaultMapper.class)
+@Configuration
+@WithSpring
+public class SpringAnnotateWithDecoratorMapperTest {
```

Comment:
Do we really need this test? I think that the most important parts are to verify that the right things are in the generated code.

Reply:
I removed the code and refactor `SpringAnnotateWithMapperTest` and related stuff to its original package. I will include ir t again if necessary.

Thank you very much for the feedback @filiphr.